Libraries/LibWeb/IndexedDB/Internal/Algorithms.cpp

```diff
@@ -1840,4 +1840,19 @@ GC::Ptr<IDBCursor> iterate_a_cursor(JS::Realm& realm, GC::Ref<IDBCursor> cursor,
     return cursor;
 }
 
+// https://w3c.github.io/IndexedDB/#clear-an-object-store
+JS::Value clear_an_object_store(GC::Ref<ObjectStore> store)
+{
+    // 1. Remove all records from store.
+    store->clear_records();
+
+    // 2. In all indexes which reference store, remove all records.
+    for (auto const& [name, index] : store->index_set()) {
+        index->clear_records();
+    }
```

Comment:
Do we also need to clear store's index set?

Reply:
No. The indecies in the store are separate from the records that exist in them